Automated spellcheck via GitHub Actions (and addressing all surfaced issues)

SeanKilleen opened the pull request:
## Describe this PR

<!-- Please describe the issue, problem or reason for the PR... A little context helps us scan the PR first, without having to dive into the code -->

* Adds automated spell-checking via a GitHub Action
* Addresses all the issues that cSpell surfaces

(Background: I've recently done this for NUnit, Marten, Playwright, Uno Platform, etc. and figured you might also find it beneficial. No worries about rejecting it if you're not interested since I didn't raise an issue first.)

## Changes

<!-- What changes have you made? Keep it brief!

- Refactored example to include new database field
- ... -->

* Adds automated spell-checking via a GitHub Action
* Addresses all the issues that cSpell surfaces

## What issue does this fix?

<!-- Link the issue this is solving, if applicable, using the issue number. This can be found to the right of the issue title, or in the url.


Fixes #1234  -->

Any open spell-check issues hopefully. 😄 But honestly in a brief glance I didn't see any that applied.

## Any other relevant information

<!-- Add any other information you deem to be relevant to the PR -->

SeanKilleen commented:
cSpell found 176 errors to address (these are usually 98% existing words that need to be added or things that back-ticks solve. I'll work to address those next.

SeanKilleen commented:
Alright, this is ready for review I think. I'll do a quick self review to point out information.

SeanKilleen commented on cSpell.json:
```diff
@@ -0,0 +1,102 @@
+{
+  "version": "0.2",
+  "language": "en",
+  "words": [
```
ℹ️ Quick note: `words` here will pop up as spelling suggestions in IDEs; `ignoreWords` will not.

SeanKilleen commented on content/300-guides/050-database/100-developing-with-prisma-migrate/165-enable-native-database-functions.mdx:
```diff
@@ -1,6 +1,6 @@
 ---
 title: Enable extensions for native database functions
-metaDecription: How to enable native database functions for projects that use Prisma Migrate.
+metaDescription: How to enable native database functions for projects that use Prisma Migrate.
```
ℹ️ Wanted to call your attention to this because I changed the Front Matter property name, assuming the thing referencing it uses the correct name. So code may need to change elsewhere to allow fixing this spelling issue.

SeanKilleen commented on content/600-about/200-prisma-docs/20-style-guide/01-writing-style.mdx:
```diff
@@ -30,9 +30,9 @@ Write in a calm, assured tone of voice. Our tone is friendly and direct, but [we
-- "color" over "colour"
-- "behavior" over "behaviour"
-- "Prisma plans to" over "Prisma plan to"
+- `color` over `colour`
```
ℹ️ I put these behind back-ticks because otherwise `behaviour` would show as a spelling error. But I didn't want to add it to the dictionary because it's a specific case we want to catch. Putting the terms in back-ticks allows them to be there without a spelling issue being raised and without compromising readability IMO.

janpio commented on content/200-concepts/050-overview/100-what-is-prisma/100-data-modeling.mdx:
```diff
@@ -8,7 +8,7 @@ metaDescription: 'Learn how data modeling with Prisma differs from data modeling
 The term _data modeling_ refers to the **process of defining the shape and structure of the objects in an application**, these objects are often called "application models". In relational databases (like PostgreSQL), they are stored in _tables_ . When using document databases (like MongoDB), they are stored in _collections_.
 
-Depending on the domain of your application, the models will be different. For example, if you're writing a blogging application, you might have models such as _blog_, _author_, _article_. When writing a carsharing app, you probably have models like _driver_, _car_, _route_. Application models enable you to represent these different entities in your code by creating respective _data structures_.
+Depending on the domain of your application, the models will be different. For example, if you're writing a blogging application, you might have models such as _blog_, _author_, _article_. When writing a cars haring app, you probably have models like _driver_, _car_, _route_. Application models enable you to represent these different entities in your code by creating respective _data structures_.
```
I think more
```suggestion
Depending on the domain of your application, the models will be different. For example, if you're writing a blogging application, you might have models such as _blog_, _author_, _article_. When writing a car sharing app, you probably have models like _driver_, _car_, _route_. Application models enable you to represent these different entities in your code by creating respective _data structures_.
```

SeanKilleen replied:
D'oh! Nice catch, thanks.

janpio commented on content/200-concepts/100-components/02-prisma-client/061-custom-validation.mdx:
```diff
@@ -53,4 +53,4 @@ The example above shows how you can create a custom type-safe `signup` function
 
 - Learn how you can organize your code better by moving the `signup` function into [a custom model](/concepts/components/prisma-client/custom-models).
-- There's an [outstanding feature request](https://github.com/prisma/prisma/issues/3528) to bake user validation into Prisma Client. If you'd like to see that happen, make sure to upvote that issue and share your use case!
+- There's an [outstanding feature request](https://github.com/prisma/prisma/issues/3528) to bake user validation into Prisma Client. If you'd like to see that happen, make sure to up-vote that issue and share your use case!
```
Hmmm https://en.wiktionary.org/wiki/upvote 🤔

SeanKilleen replied:
✅ Reverted

janpio commented on content/300-guides/050-database/100-developing-with-prisma-migrate/200-troubleshooting-development.mdx:
```diff
@@ -8,7 +8,7 @@ metaDescription: Troubleshooting issues with Prisma Migrate in a development env
 
 - [Production troubleshooting](/guides/database/production-troubleshooting)
-- [Patching / hotfixing production databases](/guides/database/patching-production)
+- [Patching / hot-fixing production databases](/guides/database/patching-production)
```
Hmmm https://en.wiktionary.org/wiki/hotfix 🤔

SeanKilleen replied:
✅ Reverted.

janpio commented on content/300-guides/200-deployment/110-deployment-guides/300-deploying-to-vercel.mdx:
```diff
@@ -17,7 +17,7 @@ The Next.js app has the following components:
 ![architecture diagram](https://res.cloudinary.com/prismaio/image/upload/v1628761153/docs/GOhQXGZ.png)
 
-The focus of this guide is showing how Prisma can be used with Next.js and Vercel. The starting point will be the [Prisma Vercel example](https://github.com/prisma/deployment-example-vercel), which has a couple of serverless REST endpoints preconfigured as [Next.js API Routes](https://nextjs.org/docs/api-routes/introduction) and a Next.js page.
+The focus of this guide is showing how Prisma can be used with Next.js and Vercel. The starting point will be the [Prisma Vercel example](https://github.com/prisma/deployment-example-vercel), which has a couple of serverless REST endpoints pre-configured as [Next.js API Routes](https://nextjs.org/docs/api-routes/introduction) and a Next.js page.
```
Hmm https://en.wiktionary.org/wiki/preconfigure 🤔

SeanKilleen replied:
✅ Reverted

janpio commented on content/400-reference/300-database-reference/01-database-features.mdx:
```diff
@@ -131,7 +131,7 @@ The following table lists common MongoDB features and describes the level of sup
 | Transactions                              | ✔️                  |                                                                                                           |
 | Indexes                                   | ✔️ with caveats     | Indexes can only be introspected if the field they refer to includes at least some data.                  |
-| Autoincrementing IDs                      | No                  |                                                                                                           |
+| Auto-incrementing IDs                      | No                  |                                                                                                           |
```
Hmm https://en.wiktionary.org/wiki/autoincrement 🤔

SeanKilleen replied:
✅ Reverted

janpio reviewed (commented):
Some nice fixes, and some things where we will need to discuss what we want from a style perspective.

SeanKilleen commented:
@janpio no worries -- this PR isn't about being "correct". Any change that a maintainer is comfortable with codifying is fine to me. All the ones you pulled out seemed reasonable and I'll make the appropriate changes to revert them and add them to the dictionary. 👍

SeanKilleen commented:
@janpio for your consideration -- I'd love to get this merged soon given that conflicts are starting to creep in. I've addressed all your comments (which were totally reasonable) and I think this is ready for review / merge.

janpio commented:
Would love to, but I am on the coding side here - @tanberry and @keerlu are actually on the docs side and can take a look.

tanberry commented:
Hi hi @SeanKilleen wow, let me take a closer look, but I'll first say Thank you very much. ;-)

tanberry commented:
Yeah for sure this tool caught a lot of typos.. I think I recognize one of mine, errrrr....

A couple of questions, @SeanKilleen . In the cSpell.json file, the list of words starting at line 5.. are these words the tool is looking for, are these "approved spelling" words, what exactly?

I agree with the vast majority of the changes, though not these ones: 
- "car sharing" needs a hyphen
- "e-commerce" no longer needs a hyphen

I'd like to get our WebDev lead to take a look and make sure she agrees about adding the two new files, and the process. We have a pretty complex set of tools already, so I want to check with her and make sure this isn't overloading tings. But it sure seems to work well, so thank you again and we will get back to you soon.

SeanKilleen commented:
@tanberry FYI the requested changes were made. As far as I can tell this one's out of my hands for now, but feel free to ping me with any questions or updates you'd like.

tanberry reviewed (approved):
This is great, thanks so very much @SeanKilleen !
